kms/kms_test.go

```diff
@@ -484,6 +484,68 @@ func TestIAMRemoveMember(t *testing.T) {
 	}
 }
 
+func TestImportEndToEnd(t *testing.T) {
+	testutil.SystemTest(t)
+
+	cryptoKeyID := fixture.RandomID()
+	t.Run("createKeyForImport", func(t *testing.T) {
+		var b bytes.Buffer
+		if err := createKeyForImport(&b, fixture.KeyRingName, cryptoKeyID); err != nil {
+			t.Fatal(err)
+		}
+
+		if got, want := b.String(), "Created key"; !strings.Contains(got, want) {
+			t.Errorf("createKeyForImport: expected %q to contain %q", got, want)
+		}
+	})
+	cryptoKeyName := fmt.Sprintf("%s/cryptoKeys/%s", fixture.KeyRingName, cryptoKeyID)
+
+	importJobID := fixture.RandomID()
+	t.Run("createImportJob", func(t *testing.T) {
+		var b bytes.Buffer
+		if err := createImportJob(&b, fixture.KeyRingName, importJobID); err != nil {
+			t.Fatal(err)
+		}
+
+		if got, want := b.String(), "Created import job"; !strings.Contains(got, want) {
+			t.Errorf("createImportJob: expected %q to contain %q", got, want)
+		}
+	})
+	importJobName := fmt.Sprintf("%s/importJobs/%s", fixture.KeyRingName, importJobID)
+
+	t.Run("checkStateImportJob", func(t *testing.T) {
+		var b bytes.Buffer
+		for !strings.Contains(b.String(), "ACTIVE") {
+			if err := checkStateImportJob(&b, importJobName); err != nil {
+				t.Fatal(err)
+			}
+		}
+	})
+
+	t.Run("importManuallyWrappedKey", func(t *testing.T) {
```

Comment:
Why is this being run in a separate goroutine?

Reply:
As above - is there some style guide rule that's violated here?

Reply:
No rule per se is being violated. However, embedding  calls to `t.Run()` creates multiple problems:

1. Adds unnecessary concurrency
2. Creates a harder-to-maintain test case